sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java

```diff
@@ -509,6 +513,41 @@ public void listKeys(HttpClient httpClient, KeyServiceVersion serviceVersion) {
         });
     }
 
+    /**
+     * Tests that an existing key can be released.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void releaseKey(HttpClient httpClient, KeyServiceVersion serviceVersion) {
+        // TODO: Remove assumption once Key Vault allows for creating exportable keys.
+        Assumptions.assumeTrue(isManagedHsmTest);
```

Comment:
This is client-side validation and you should let the service error. We don't want tight coupling between SDK and service (it's in our guidelines). Only in rare cases - like null checks for required URL path parameters - do we do client-side validation of service client methods.

Reply:
We do let the service error when it comes down to the public APIs. I added this to a test case just to avoid having it break the live tests. Should I remove it?

Reply:
I recommend it, yes. If/when the service does change - especially unexpectedly - you'll know when it doesn't match the recordings, assuming you validate your request bodies like we do in .NET by default.